runtime/lua/vim/lsp/util.lua

```diff
@@ -226,6 +226,141 @@ end
 -- function M.glob_to_regex(glob)
 -- end
 
+function table.slice(tbl, first, last)
+  local sliced = {}
+
+  for i = first or 1, last or #tbl do
+    sliced[#sliced+1] = tbl[i]
+  end
+
+  return sliced
+end
+
+local function first_difference(old_buffer, new_buffer)
+	local line_count = math.min(#old_buffer, #new_buffer)
+	if line_count == 0 then return { 0, 0} end
+	local start_line_idx = line_count
+	for i = 1, line_count do
+		if old_buffer[i] ~= new_buffer[i] then
+			start_line_idx = i
+			break
+		end
+	end
+	if start_line_idx > line_count then
```

Comment:
Is `start_line_idx` not always smaller or equal to `line_count`?

Reply:
In the original code it was >=, so it was an optimization to return early. I may have broken this in my attempt to fix all the off by one errors.

Reply:
I think it's fixed now, hopefully no regressions :) None that I saw at least testing rust analyzer

Reply:
Ok, it's not fixed. I had to undo this as it was causing issues with full buf ranges.